vmm/src/lib.rs

```diff
@@ -1125,6 +1130,19 @@ impl Vmm {
         Ok(())
     }
 
+    #[cfg(target_arch = "x86_64")]
+    fn get_serial_device(&self) -> Option<Arc<Mutex<RawIOHandler>>> {
+        Some(self.legacy_device_manager.stdio_serial.clone())
+    }
+
+    #[cfg(target_arch = "aarch64")]
+    fn get_serial_device(&self) -> Option<&Arc<Mutex<RawIOHandler>>> {
+        self.mmio_device_manager
+            .as_ref()
+            .unwrap()
+            .get_raw_io_device(DeviceType::Serial, "uart")
```

Comment:
I don't personally like that we're building a "uart" `String` object that we then use in a hashmap search for each `Serial` input/output.